### PR TITLE
[2361] - Basic course page

### PR DESCRIPTION
**Pull request by TobyRet**
### Context
Creation of a very basic course page just to establish communication between Find and MCBE.

### Changes proposed in this pull request
In addition to creation of course page this work includes: 

- Creation of MCBE client and API serialization
- Creation of basic models (Provider, Site, Recruitmentcycle)
- Testing setup e.g. request stubbing, page object model, factories
 
### Guidance to review
_Probably quite important_ - you will notice that a lot of this work is inspired by Manage Courses Frontend. I've tried to refrain from simply lifting and shifting code in favour of adding the bare minimum. This might be a good opportunity to reflect on Manage Courses Front End's implementation and decide what aspects we model (or try to do differently). **Worth a wider team discussion?**

screen

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -13,4 +13,6 @@
   get "/404", to: "errors#not_found", via: :all
   get "/422", to: "errors#unprocessable_entity", via: :all
   get "/500", to: "errors#internal_server_error", via: :all
+
+  get "/courses/:provider_code/:course_code", to: "courses#show", as: "course"
```
Comment (defong) [inline config/routes.rb]: is that right 

https://find-postgraduate-teacher-training.education.gov.uk/course/T92/X130

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CoursesController < ApplicationController
+  def show
+    @course = Course
+                .where(recruitment_cycle_year: 2020)
```
Comment (defong) [inline app/controllers/courses_controller.rb]: 2020 use it from config please

File: app/views/courses/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<%= content_for :page_title, "#{@errors.present? ? "Error: " : ""}#{@course.name} - Courses" %>
```
Comment (defong) [inline app/views/courses/show.html.erb]: error?

**Comment by dankmitchell** [comment]
Should we move the config setup into a different PR before this one?

**Comment by TobyRet** [comment]
[Build basic course page (Find)](https://trello.com/c/gTTzNhca/2361-build-basic-course-page-find)

**Pull request merged**